File: LeetCode/python3/153.py
class Solution:
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        start, end = 0, len(nums) - 1
        while start < end - 1:
            mid = start + (end - start) // 2
            if nums[mid] < nums[start] and nums[mid] < nums[end]:
                end = mid
            elif nums[mid] > nums[start] and nums[mid] > nums[end]:
                start = mid
            else:
                return nums[0]
        return min(nums[start], nums[end])

File: LeetCode/cpp/153.cpp
class Solution {
public:
    int findMin(vector<int> nums) {
        int i = 0, j = nums.size() - 1, mid = 0;
        if (j == -1)
            return 0;
        while (i < j - 1) {
            mid = i + (j - i) / 2;
            if (nums[mid] > nums[i] && nums[mid] > nums[j])
                i = mid;
            else if (nums[mid] < nums[i] && nums[mid] < nums[j])
                j = mid;
            else
                return nums[i];
        }
        return min(nums[i], nums[j]);
    }
};